codonPython/nhsNumberGenerator.py

```diff
@@ -41,9 +41,14 @@ def nhsNumberGenerator(to_generate: int, random_state: int = None)->list:
         digits = [int(digit) for digit in str(number)]
         # Apply weighting to digits
         weighted_digits = [(10 - index) * digit for (index, digit) in enumerate(digits)]
-        # Sum of all weighted digits must be a multiple of 11 to be valid.
-        if sum(weighted_digits) % 11 == 0:
-            # Add check digit to valid number
-            number = int(str(number) + "0")
-            generated.append(number)
-    return generated
\ No newline at end of file
+        # Validity is based on the check digit, which can't be 10   
+        remainder = sum(weighted_digits) % 11
+        check_digit = 11 - remainder
+        if check_digit == 10:
+            continue
```

Comment:
This is not visited during testing.

Reply:
I've extended the numbers generated in testing so they are.